File: inc_file.pl
#!/usr/bin/perl

#full language support for code2html
#ada, ada95, awk, c, c++, cc, cpp, cxx, gpasm, groff, html, java, javascript, js, lisp, m4, make, makefile, pas, pascal, patch, perl, plain, pov, povray, python, ruby, sh, shellscript, sql

#language support we are using:
sub file_match {
	my $string;
	$string = $_[0];
	if ( $string =~ /(<FILE=")(.*?)(">)/s ) { 
		return 1;
	}
	else {return 0;}
}

sub file_to_text {
	$fname=$_[0];
	my $txt;
	$txt='';
	if ( !open INPUTDATA, $fname ) {
		warn "cannot open $fname: $!";
	} else {
	   while (<INPUTDATA>) {
		  $txt = $txt . $_;
   	}
   }
	return $txt;
}


sub file_replace {
	my $string;
	$string = $_[0];
	if ($string =~ /(.*)(<FILE=")(.*?)(">)(.*)/s ) {
		$before = $1;
		$fname = $3;
		$after = $5;
	}
	
	$file_content=&file_to_text($fname);
   
	return $before . $file_content . $after;
}


$tmp='';
while ( <> ) {
	$tmp = $tmp . $_;
}

  $count = 0;
  while( &file_match($tmp) ) {
	  $tmp=&file_replace($tmp);
	  $count=$count + 1;
	  if ($count > 1000) {die "something went wrong!"}
  }
  
print $tmp;

File: theme1/bin/inc_file.py
#!/usr/bin/env python3
import re
import sys


def file_match(text: str) -> bool:
    return re.search(r'(<FILE=")(.*?)(">)', text, flags=re.DOTALL) is not None


def file_to_text(fname: str) -> str:
    try:
        with open(fname, "r") as input_file:
            return input_file.read()
    except OSError as exc:
        print(f"cannot open {fname}: {exc}", file=sys.stderr)
        return ""


def file_replace(text: str) -> str:
    match = re.match(r'(.*)(<FILE=")(.*?)(">)(.*)', text, flags=re.DOTALL)
    if not match:
        return text
    before, fname, after = match.group(1), match.group(3), match.group(5)
    file_content = file_to_text(fname)
    return f"{before}{file_content}{after}"


def main() -> None:
    content = sys.stdin.read()
    count = 0
    while file_match(content):
        content = file_replace(content)
        count += 1
        if count > 1000:
            raise RuntimeError("something went wrong!")
    sys.stdout.write(content)


if __name__ == "__main__":
    main()
